cmd/registry/registryList.go

```diff
@@ -66,62 +78,33 @@ func NewCmdRegistryList() *cobra.Command {
 				}
 			}
 			existingNodes = client.NodeFilterByRoles(existingNodes, []k3d.Role{k3d.RegistryRole}, []k3d.Role{})
+
 			// print existing registries
-			if len(existingNodes) > 0 {
-				printNodes(existingNodes, headersOff)
+			headers := &[]string{}
+			if !registryListFlags.noHeader {
+				headers = &[]string{"NAME", "PORT", "ROLE", "CLUSTER"} // TODO: add status
 			}
+
+			util.PrintNodes(existingNodes, registryListFlags.output,
+				headers, util.NodePrinterFunc(func(tabwriter *tabwriter.Writer, node *k3d.Node) {
+					cluster := "*"
+					if _, ok := node.Labels[k3d.LabelClusterName]); ok {
+						cluster = node.Labels[k3d.LabelClusterName])
+					}
+					fmt.Fprintf(tabwriter, "%s\t%s\t%s\n",
+						strings.TrimPrefix(node.Name, "/"),
+						string(node.Role),
+						cluster
+				}))
 		},
```

Comment:
Some fixes to my previous suggestion, this time tested in vscode:
```suggestion
			util.PrintNodes(existingNodes, registryListFlags.output,
				headers, util.NodePrinterFunc(func(tabwriter *tabwriter.Writer, node *k3d.Node) {
					cluster := "*"
					if _, ok := node.Labels[k3d.LabelClusterName]; ok {
						cluster = node.Labels[k3d.LabelClusterName]
					}
					fmt.Fprintf(tabwriter, "%s\t%s\t%s\n",
						strings.TrimPrefix(node.Name, "/"),
						string(node.Role),
						cluster,
					)
				}),
			)
		},
```